core/media_streamer.go

```diff
@@ -102,6 +103,34 @@ func (ms *mediaStreamer) NewStream(ctx context.Context, id string, reqFormat str
 	return s, nil
 }
 
+func (ms *mediaStreamer) TranscodeDownload(ctx context.Context, mf *model.MediaFile, format string, bitrate int, w io.Writer) error {
+	if bitrate == 0 {
+		bitrate = mf.BitRate
+	}
+
+	job := &streamJob{
+		ms:      ms,
+		mf:      mf,
+		format:  format,
+		bitRate: bitrate,
+	}
+
+	r, error := ms.cache.Get(ctx, job)
```

Comment:
Please don't use `error` as a variable name as it colides with the builtin `error` interface.